modules/apps/rate-limiting/keeper/packet.go

```diff
@@ -122,17 +118,20 @@ func ParseDenomFromSendPacket(packet transfertypes.FungibleTokenPacketData) (den
 //	        -> Remove Prefix: transfer/channel-Z/ujuno
 //	        -> Hash:          ibc/...
 func ParseDenomFromRecvPacket(packet channeltypes.Packet, packetData transfertypes.FungibleTokenPacketData) (denom string) {
```

Comment:
I feel like a lot of this is logic that should either already be in ICS20Transfer already and re-used across all middleware. We should look for those and try to generalize some of these.